src/fabric/annealing.rs

```diff
@@ -38,11 +45,19 @@ impl Fabric {
             .enumerate()
             .map(|(index, joint)| JointIncident::new(index, joint.location)).collect();
         for interval @ Interval { alpha_index, omega_index, .. } in self.interval_values() {
-            incidents[*alpha_index].add(interval);
-            incidents[*omega_index].add(interval);
+            incidents[*alpha_index].add_interval(interval);
+            incidents[*omega_index].add_interval(interval);
         }
         incidents
     }
+
+    fn interval_map(&self) -> HashMap<(usize, usize), Interval> {
+        let mut hashmap = HashMap::new();
+        for interval in self.interval_values() {
+            hashmap.insert(interval.key(), *interval);
+        }
+        hashmap
+    }
```

Comment:
```suggestion
        self.interval_values()
          .into_iter()
          .map(|interval| (interval.key(), *interval))
          .collect()
    }
```
`HashMap` implements `FromIterator<Item=(K, V)>`

Reply:
I haven't compiled this to check that it works BTW